README.md

```diff
@@ -1,17 +1,38 @@
+# Synopsis
+```
+$ nova-password -h
+Get the admin password for an OpenStack server
+
+Usage:
+  nova-password <server-name>|<server-id> [<server-name>|<server-id>...] [flags]
+
+Flags:
+  -d, --debug                     print out request and response objects
+  -h, --help                      help for nova-password
+  -i, --private-key-path string   a path to the RSA private key (PuTTY and OpenSSH formats) (default "/Users/uuu/.ssh/id_rsa")
+  -v, --version                   print the nova-password version
+
+```
 # Usage
 
-## Prerequisites
+- source the `openrc` of the project where the server is running in the current shell to load the relevant ENV vars
 
-* The private key corresponding to the public key, used to create a compute instance, is required
-* Only RSA PKCS #1 v1.5 is supported by OpenStack
+
+## Prerequisites
+- The user's (admin) password is provisioned when the server is created. We assume this hasn't been altered.
+- During creation, the password is encrypted using your **public** SSH key, which is in your User Profile. 
+- The **private** key corresponding to that public key, is used to decrypt the password.
+  It is assumed to be your default private key, but you can specify another using `-i`. 
+- Only RSA PKCS #1 v1.5 is supported by OpenStack (**elaborate**)
```

Comment:
Elaborating will significantly explode the doc. Short answer - because RSA supports public key encryption: https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#encryption

Reply:
This "_Only RSA PKCS_" Prerequisite (or constraint) is on OpenStack, not on the tool. 

Adding this statement here just confuses the user: all he wants is to know if he can use `id_rsa` and/or PuTTY's `putty.ppk`

Reply:
changes on branch patch-1

Reply:
I;d keep this statement. Users may create ed25519, dsa or ecdsa SSH keypairs. These keys won't work for them and users will be confused.